main.go

```diff
@@ -363,20 +374,23 @@ func getFromMetadata(urlStr string) ([]byte, error) {
 		},
 	}
 	resp, err := client.Do(req)
+	defer resp.Body.Close()
 	if err != nil {
 		return nil, fmt.Errorf("failed communicating with metadata server: %w", err)
 	}
 	body, err := ioutil.ReadAll(resp.Body)
-	resp.Body.Close()
 	if err != nil {
 		return nil, fmt.Errorf("failed reading from metadata server: %w", err)
 	}
+	if statusOk := resp.StatusCode >= 200 && resp.StatusCode < 300; !statusOk {
+		return nil, fmt.Errorf("failed reading from metadata server with Non-OK status code: %d", resp.StatusCode)
+	}
```

Comment:
Maybe something like:

```golang
if code := resp.StatusCode; code < 200 || code > 299 {
	return nil, fmt.Errorf("metadata server returned status code %d for url %q", code, url)
}
```

I'm OK with the conditional being `!(code >= 200 && code < 300)` as well. But I feel the above one is simpler. Up to you.